File: code_forces/144A/a.py
import sys

def main():
    sys.stdin = open('input.txt', 'r')

    n = int(input())

    soldiers = list(map(int, input().split()))
    ans = 0

    while soldiers[0] != max(soldiers):
        index = soldiers.index(max(soldiers))
        soldiers[index], soldiers[index-1] = soldiers[index-1], soldiers[index]
        ans+=1


    while soldiers[len(soldiers)-1] != min(soldiers):
        soldiers.pop()
        ans+=1


    print(ans)


if __name__ == '__main__':
    main()

File: code_forces/144A/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    
    freopen("input.txt", "r", stdin);
    int n, ans=0;

    vector<int> soldiers;

    cin>>n;

    for(int i=0; i<n; i++) {
        int temp;
        cin>>temp;
        soldiers.push_back(temp);
    }

    int x = *max_element(soldiers.begin(), soldiers.end());

    auto min_max = minmax_element(soldiers.begin(), soldiers.end());
    int min_ele = *min_max.first;
    int max_ele = *min_max.second;
    while(soldiers[0] != max_ele) {
        int max_index = max_element(soldiers.begin(), soldiers.end()) - soldiers.begin();
        swap(soldiers[max_index], soldiers[max_index-1]);
        ans++;
    }

    while(soldiers[n-1] != min_ele) {
        soldiers.pop_back();
        n--;
        ans++;
    }

    cout<<ans;
    return 0;
}